### DISCUSSION TITLE
Postgres authentication failed on RPI

### DISCUSSION THREAD

**Discussion by OevreFlataeker** [nzymedefense/nzyme#545] [Questions] [answered]
I am a new user to nzyme and didn't use postgresql before. 
I followed the installation instructions for RPI [here](https://www.nzyme.org/docs/installation-configuration/install-raspberry-pi-os) and set up the nzyme database accordingly. OS is Debian "buster" according to /etc/os-release. Postgres is version 11.12

In the command ''create user nzyme with encrypted password 'YOUR_PASSWORD_HERE';'' I've set my password "s00pers3cret". In bash I created the sha256 of the same pw and added it to the nzyme. config.

When nzyme starts up, it complains it cannot connect to the database. I checked the postgresql logs ("PW invalid for user nzyme" and a reference to pg_hba.conf. In that config I saw that the engine seems to work with md5 instead of sha256. (/etc/postgresql/pg_hba.conf). 

host    all             all             127.0.0.1/32            md5

When I changed that to ''scram-sha-256'' I get another error on restart saying that no valid SCRAM verifier was set for the user.

It seems postgresql does not like nzyme using sha256 for authentication? Can I force nzyme to use md5 for testing purposes?

**Comment by lennartkoopmann** [comment, marked as answer]
You must not `sha256` the password in the nzyme configuration. It goes in there unhashed.

**Reply by OevreFlataeker** [reply]
Thanks I have mistaken ''general.admin_password_hash'' as the PostgreSQL password
